### ISSUE TITLE
Cherry-pick and merge upstream changes

### ISSUE THREAD

**Issue by quozl** [sugarlabs/grid-paint-activity#5]
sugarlabs/grid-paint-activity repository was uploaded from a bundle, and does not contain the original commits.

Two upstream repositories exist with useful changes since the bundle;

* https://github.com/llaske/Gridpaint.activity
* https://github.com/llaske/sugarizer/tree/dev/activities/Gridpaint.activity

The first repository also contains the original commits.

Both repositories contain commits that avoid using WebKit directly, instead the Sugar Toolkit's `sugar-activity-web` starts WebKit on behalf of the activity.

**Comment by llaske** [comment]
What do you mean by 

> avoid using WebKit directly

I there a difference between https://github.com/sugarlabs/sugar-web/blob/master/env.js#L49 
and https://github.com/llaske/sugarizer/blob/dev/activities/Gridpaint.activity/lib/sugar-web/env.js#L80 ?

**Comment by quozl** [comment]
Indeed.  "_using WebKit directly_" means that activity has an embedded [Python wrapper](https://github.com/sugarlabs/grid-paint-activity/blob/master/activity.py) that has recently been maintained, although now I see the metadata contains [`exec = sugar-activity-web`](https://github.com/sugarlabs/grid-paint-activity/blob/master/activity/activity.info#L5) which means the wrapper should have been ignored?  I'm confused.  Still, the code in this repository is older than yours.

**Comment by quozl** [comment]
@iqraceme, would you be able to get that change to env.js into sugar-web?  sugar-web has aged without a maintainer.  I don't know enough about the JavaScript environment to understand what it means, but you probably do.

**Comment by cemeiq** [comment]
Yeah, I'll check it and get that change to env.js into sugar-web.
